include/boost/geometry/algorithms/hausdorff_distance.hpp

```diff
@@ -0,0 +1,215 @@
+        <
+            typename point_type<Linestring1>::type,
+            typename point_type<Linestring2>::type,
+            Strategy
+        >::type apply(Linestring1 const& ls1, Linestring2 const& ls2, Strategy const& strategy)
+    {
+        
+        typedef typename distance_result
+            <
+                typename point_type<Linestring1>::type,
+                typename point_type<Linestring2>::type,
+                Strategy
+            >::type result_type;
+        typedef typename boost::range_size<Linestring1>::type size_type;
+        
+        result_type dis_max=0,dis_min;
+        boost::geometry::detail::throw_on_empty_input(ls1);
+        boost::geometry::detail::throw_on_empty_input(ls2);
+        size_type  a = boost::size(ls1);
+        size_type  b = boost::size(ls2);
+
+        //Computing the HausdorffDistance
+        for(size_type i=0;i<a;i++)
+        {
+            bool is_dis_min_set = false;
+                for(size_type j=0;j<b;j++)
+                {
+                    result_type dis_temp = geometry::distance(range::at(ls1, i), range::at(ls2, j), strategy);
+                    if(dis_temp < dis_max)
+                        break; //Early Break
+                    if(!is_dis_min_set || dis_temp < dis_min)
+                    {    
+                        dis_min=dis_temp;
+                        is_dis_min_set = true;
+                    }
+                }
+
+            if (dis_min > dis_max && is_dis_min_set)
+            {
+              dis_max = dis_min;
+            }
+        }
+        
+        return dis_max;
+    }
+};
+
+struct linestring_multi_linestring
+{
+    template <typename Linestring, typename Multi_linestring, typename Strategy>
+    static inline typename distance_result
+        <
+            typename point_type<Linestring>::type,
+            typename point_type<Multi_linestring>::type,
+            Strategy
+        >::type apply(Linestring const& ls, Multi_linestring const& mls, Strategy const& strategy)
+    {
+        
+        typedef typename distance_result
+            <
+                typename point_type<Linestring>::type,
+                typename point_type<Multi_linestring>::type,
+                Strategy
+            >::type result_type;
+        typedef typename boost::range_size<Linestring>::type size_type1;
+        typedef typename boost::range_size<Multi_linestring>::type size_type2;
+        
+        boost::geometry::detail::throw_on_empty_input(ls);
+        boost::geometry::detail::throw_on_empty_input(mls);
+        size_type1  a = boost::size(ls);
+        size_type2  b = boost::size(mls);
+
+        result_type haus_dis_max=0;
+        //Computing the HausdorffDistance
+        for(size_type1 i=0;i<a;i++)
+        {
+                result_type dis_max=0;
+                for(size_type2 j=0;j<b;j++)
+                {
+                    result_type dis_min;
+                    bool is_dis_min_set = false;
```

Comment:
This part of the code is very similar to the algorithm for Ls-Ls. To avoid duplication of code you should make a helper function from this part and call it inside the algorithm for Ls-Ls and Ls-MLs. So a function taking two ranges of points, PtPtDistanceStrategy and the current max distance (which will be updated by the function, so both input and output variable).

And later do the same with MLs-MLs because a part of algorithm there will be the same as this algorithm for Ls-MLs.

You could name these helper functions `range_range()`, and `range_multirange()` because they could also be used for MultiPoint-MultiPoint so you should use some generic (not geometry specific) name.

You could also have `point_range()` function (this would be the most inner loop you have in algorithms). And you could call it inside a loop in `range_range()` and directly in Point-MultiPoint algorithm.

Reply:
Or you could simply call `linestring_linestring()` here. Maybe it's even better.

Note that you could use the same function for MultiPoint-MultiPoint, so still renaming it to `range_range()` makes sense.

Reply:
I have Updated hausdorff_distance, now I am calling linestring_linestring().